ganttproject/src/net/sourceforge/ganttproject/action/resource/AssignmentToggleAction.java

```diff
@@ -0,0 +1,92 @@
+/*
+GanttProject is an opensource project management tool. License: GPL3
+Copyright (C) 2019 GanttProject Team
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 3
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package net.sourceforge.ganttproject.action.resource;
+
+import net.sourceforge.ganttproject.action.GPAction;
+import net.sourceforge.ganttproject.gui.UIFacade;
+import net.sourceforge.ganttproject.gui.UIFacade.Choice;
+import net.sourceforge.ganttproject.resource.AssignmentContext;
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.ResourceAssignmentMutator;
+import net.sourceforge.ganttproject.task.Task;
+import net.sourceforge.ganttproject.util.StringUtils;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Action that adds/removes the assignment of a task to a resource
+ */
+public class AssignmentToggleAction extends GPAction {
+  private final HumanResource myHumanResource;
+
+  private final UIFacade myUIFacade;
+
+  private final ResourceAssignmentMutator myMutator;
+  private final Task myTask;
+  private ArrayList<ResourceAssignment> myAssignments;
+
+  public AssignmentToggleAction(HumanResource hr, UIFacade uiFacade, Task task) {
+    super(hr.getName(), IconSize.TOOLBAR_SMALL);
+    myHumanResource = hr;
+    myUIFacade = uiFacade;
+    myTask = task;
+    myMutator = task.getAssignmentCollection().createMutator();
+    myAssignments = new ArrayList<ResourceAssignment>(Arrays.asList(myTask.getAssignmentCollection().getAssignments()));
+  }
+
+  private void createAssignment() {
+      ResourceAssignment newAssignment = myMutator.addAssignment(myHumanResource);
+      newAssignment.setLoad(100);
+
+      boolean coord = false;
+      if (myAssignments.isEmpty())
```

Comment:
I don't think that it makes sense to add such logic with coordinator field here. The matter is that who is the corrdinator depends on the order of adding resources, and, unlike the case when it happens in the table, you don't see the effect.

Reply:
Removed the coordinator-related logic